### DISCUSSION TITLE
Anyone running this on a Synology NAS?

### DISCUSSION THREAD

**Discussion by anivdas** [jasonacox/Powerwall-Dashboard#79] [Q&A]
Our Tesla Solar+Powerwall install date is scheduled for next week and this dashboard looks like everything I could ever want. Is there a docker stack guide I can use to set this up on my Synology NAS Docker?

**Comment by jasonacox** [comment]
I know it is possible and we have had several report they are doing this.  It may work without any change, but if you have any issue, you might want to reference this: https://github.com/jasonacox/Powerwall-Dashboard/issues/22#issuecomment-1110099800

**Reply by ripclaw786** [reply]
Thanks, Jason.  I also have a Synology NAS with docker installed.  The main docker directory resides in **/volume1/docker**.

Do I need to change anything in the powerwall.yml script, or can I just issue the "git clone" command immediately after I've ssh'd into my Synology NAS and then run the setup.sh script?

Thanks in advance.

**Reply by jasonacox** [reply]
If we did it correctly, you should be able to `git clone` and `setup.sh` without any changes.  I don't have a Synology to test so I would love your feedback and help to adjust if we need to make any changes.

**Reply by jasonacox** [reply]
Found this on Reddit: https://www.reddit.com/r/TeslaSolar/comments/yej96t/comment/iug1ls8/?utm_source=share&utm_medium=web2x&context=3 and modified it to work with the latest version of the Dashboard that includes the compose.env file:

Some linux knowledge will be helpful.

1.Enable SSH on Synology Nas.

2.Install git on Synology Nas.

3.You need to download an ssh client (eg. bitvise), and then connect to Synology nas with SSH. Once inside, type

```
cd volume1/docker
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
cd Powerwall-Dashboard
./setup.sh
```

Enter your details (including time zone, powerwall ip , user id, pwd etc). In my case, the Grafana wouldnt load up, so I had to make a tiny change

```
id-u and you will get user id (1026 in my case)
nano compose.env
```

* Scroll down and change `GRAFANAUSER="1000:1000"` to be `GRAFANAUSER="1026"` or whatever user id you get above
* CTRL +X to close and save the file
* Start the containers: `./compose-dash.sh up -d`
* Navigate to your nas address:9000 and proceed according to the github instructions.

**Reply by jeffkruczynski** [reply]
Also, if you haven't already created a firewall entry for your docker container network(s), see [this discussion topic](https://github.com/jasonacox/Powerwall-Dashboard/discussions/122) for notes on that.

**Reply by ripclaw786** [reply]
> Found this on Reddit: https://www.reddit.com/r/TeslaSolar/comments/yej96t/comment/iug1ls8/?utm_source=share&utm_medium=web2x&context=3 and modified it to work with the latest version of the Dashboard that includes the compose.env file:
> 
> Some linux knowledge will be helpful.
> 
> 1.Enable SSH on Synology Nas.
> 
> 2.Install git on Synology Nas.
> 
> 3.You need to download an ssh client (eg. bitvise), and then connect to Synology nas with SSH. Once inside, type
> 
> ```
> cd volume1/docker
> git clone https://github.com/jasonacox/Powerwall-Dashboard.git
> cd Powerwall-Dashboard
> ./setup.sh
> ```
> 
> Enter your details (including time zone, powerwall ip , user id, pwd etc). In my case, the Grafana wouldnt load up, so I had to make a tiny change
> 
> ```
> id-u and you will get user id (1026 in my case)
> nano compose.env
> ```
> 
> * Scroll down and change `GRAFANAUSER="1000:1000"` to be `GRAFANAUSER="1026"` or whatever user id you get above
> * CTRL +X to close and save the file
> * Start the containers: `./compose-dash.sh up -d`
> * Navigate to your nas address:9000 and proceed according to the github instructions.

Many thanks, Jason!  The above helped me immensely.

Other than changing the user ID in the powerwall.yml file, I also had to change the IP address of my Tesla Gateway 2 in the powerwall.env file, from the default 192.168.91.1 to what it was on my LAN.
